docs/feature/subnet-scarcity/phase-3/1-watchpods.md

```diff
@@ -1,4 +1,157 @@
+
+Additionally, any read-only requests (`Get`, `List`, `Watch`) that CNS makes to Kubernetes using a cache-aware client will hit the local Cache instead of querying the remote API Server. This means that the only requests leaving CNS to the API Server for this Pod Watcher will be the Reflector's List and Watch.
+
+#### Server-side filtering
+To further reduce API Server load and traffic, CNS can use an available [Field Selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/) for Pods: [`spec.nodeName=<node>`](https://github.com/kubernetes/kubernetes/blob/691d4c3989f18e0be22c4499d22eff95d516d32b/pkg/apis/core/v1/conversion.go#L40). Field selectors are, like Label Selectors, [applied on the server-side](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering) to List and Watch queries to reduce the dataset that is returned from the API Server to the Client. 
+
+By restricting the Watch to Pods on the current Node, the traffic generated by the Watch will be proportional to the number of Pods on that Node, and will *not* scale in relation to either the number of Nodes in the cluster or the total number of Pods in the cluster.
+
+### Controller-runtime
+To make setting up the filters, SharedInformers, and cache-aware client easy, we will use [`controller-runtime`](https://github.com/kubernetes-sigs/controller-runtime) and create a Pod Reconciler. A controller already exists for managing the `NodeNetworkConfig` CRD lifecycle, so the necessary infrastructure (namely, a Manager) already exists in CNS.
+
+To create a filtered Cache during the Manager instantiation, the existing `nodeScopedCache` will be expanded to include Pods:
+
+```go
+import (
+  v1 "k8s.io/api/core/v1"
+  "k8s.io/apimachinery/pkg/fields"
+  "sigs.k8s.io/controller-runtime/pkg/cache"
+  //...
+)
+//...
+nodeName := "the-node-name"
+// the nodeScopedCache sets Selector options on the Manager cache which are used
+// to perform *server-side* filtering of the cached objects. This is very important
+// for high node/pod count clusters, as it keeps us from watching objects at the
+// whole cluster scope when we are only interested in our Node's scope.
+nodeScopedCache := cache.BuilderWithOptions(cache.Options{
+  SelectorsByObject: cache.SelectorsByObject{
+    // existing options
+    //...,
+    &v1.Pod{}: {
+      Field: fields.SelectorFromSet(fields.Set{"spec.nodeName": nodeName}),
+    },
+  },
+})
+//...
+manager, err := ctrl.NewManager(kubeConfig, ctrl.Options{
+    // existing options
+    //...,
+    NewCache:           nodeScopedCache,
+})
+```
+
+After the local Cache and ListWatch has been set up correctly, the Reconciler should use the Manager-provided Kubernetes API Client within its event loop so that reads hit the cache instead of the real API.
```

Comment:
It doesn't matter: when we get a Pod event, we re-list Pods. We don't go looking for that specific Pod, only query for the total number of known Pods. 

If we were looking for a specific Pod, it probably still doesn't matter because we would abort the reconcile with a "NotFound" and the Event would be requeued for us to try again later.

Even that is very unlikely though, as for us to get an Event, the resource must have been sent to the Watch and copied by the Reflector in to the Informer, which pushes the resource in to the local cache and dispatches the Event indicating the change. Ref the included diagram.